### PR TITLE
Fix db/db++/pse evaluation bugs and results, rescale polygons in postproc by default using `shape_list`; New arg parser to support changing config arg from CLI (e.g. `-o eval.ckpt_load_path=db_r50.ckpt`); Impr code format and guideline docs

### PR DESCRIPTION
**Pull request by SamitHuang**
…proc by default using , make comparision fair; Impr code format and guideline docs

Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

More text detections models are on their way like pse, east, but their evaluation methods are different in rescaling and the intermediate and end rounding for polygons. And the resizing/scaling ops they used are different in data transformation, e.g., GridResize used math.ceil in 32 division for resizing but DetResize used rounding, which leader to difficulty in yielding fair and consistent comparisons between models. 

To solve it,
1) Align different resizing method for detection evaluation and use the same protocol to record `shape_list` for rescaling later
2) Align different det postprocessing modules based on det base postprocessing.
3) Rescaling the postprocessed results back to the original image space for all text detection models in evaluation. NO rounding for the results on the transformed image space, and DO rouding in the end of rescaling for better precision. 

To test the changes
1) Re-eval DB/DB++ with the new aligned resizing methods. results are as expected.
2) Update db/db++ eval results, most of them get improved.

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

**Comment by zhtmike** [comment]
The code formatting change and the functional change is better to be separated into two commit; and once the pre-commit hook of formatting is finished, the formatting change of the entire code is better to be in one time

File: mindocr/data/transforms/transforms_factory.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 
 # TODO: use class with __call__, to perform transformation
-def create_transforms(transform_pipeline, global_config=None):
+def create_transforms(transform_pipeline: list, global_config: dict=None):
```
Comment (zhtmike) [inline mindocr/data/transforms/transforms_factory.py:17]: this kind of type hint is only support in python 3.9 or above, for 3.7/3.8, better to use from `typing.List[Any]` and `typing.Dict[Any, Any]`, or `from __future__ import annotations`

File: tools/eval.py [hunk 2]
```diff
@@ -17,96 +18,118 @@
...
 from mindocr.postprocess import build_postprocess
 from mindocr.metrics import build_metric
-from mindocr.utils.callbacks import Evaluator
+from mindocr.utils.evaluator import Evaluator
+from mindocr.utils.logger import get_logger
+
 
 def main(cfg):
     # env init
+    #ms.set_context(mode=cfg.system.mode, device_id=7)
```
Comment (zhtmike) [inline tools/eval.py:27]: better to remove commented debugging code

File: mindocr/data/transforms/rec_transforms.py [hunk 3]
```diff
@@ -32,6 +32,7 @@ def __init__(self,
                 use_space_char=False,
                 blank_at_last=True,
                 lower=False,
+                **kwrags,
```
Comment (zhtmike) [inline mindocr/data/transforms/rec_transforms.py:35]: typo

File: mindocr/data/transforms/transforms_factory.py (hunk 1 above)
Comment (SamitHuang) [reply mindocr/data/transforms/transforms_factory.py:17]: > The code formatting change and the functional change is better to be separated into two commit; and once the pre-commit hook of formatting is finished, the formatting change of the entire code is better to be in one time

True. Only format part of it.

File: mindocr/data/transforms/rec_transforms.py (hunk 3 above)
Comment (SamitHuang) [reply mindocr/data/transforms/rec_transforms.py]: fixed. thanks.

File: tools/eval.py (hunk 2 above)
Comment (SamitHuang) [reply tools/eval.py]: fixed. thanks.

File: mindocr/data/transforms/transforms_factory.py (hunk 1 above)
Comment (zhtmike) [reply mindocr/data/transforms/transforms_factory.py]: Seems without claiming the internal parameter of the generic type, `xxx: list` is also ok

**Review by zhtmike — APPROVED** [review]

**Review by Songyuanwei — APPROVED** [review]

File: mindocr/postprocess/det_base_postprocess.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+        #print('DEBUG: before rescale, poly 0: ', polygons[0], 'shape list: ', shape_list[0])
+        if isinstance(polygons, np.ndarray):
+            polygons = np.round(polygons / scale_w_h)
+            polygons[...,0] = np.clip(polygons[..., 0], 0, src_w - 1)
+            polygons[...,1] = np.clip(polygons[..., 1], 0, src_h - 1)
+        else:
+            polygons = [np.round(poly / scale_w_h) for poly in polygons]
+            for i, poly in enumerate(polygons):
+                polygons[i][:,0] = np.clip(poly[:, 0], 0, src_w - 1)
```
Comment (VictorHe-1) [inline mindocr/postprocess/det_base_postprocess.py:119]: src_w and src_h are not defined

**Review by HaoyangLee — APPROVED** [review]

**Review by HaoyangLee — APPROVED** [review]

**Review by HaoyangLee — APPROVED** [review]

**Review by VictorHe-1 — APPROVED** [review]

**Pull request merged**